packages/react-atlas-core/src/Icon/Icon.js

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import PropTypes from "prop-types";
+import cx from "classnames";
+
+/**
+ * Master Icon component. Used as a stand alone component or in conjunction with button and avatar
+ */
+class Icon extends React.PureComponent {
+  constructor(props) {
+    super(props);    
+
+    if (!this.props.icon) {
+      console.warn(
+        "You are attempting to use an icon without passing in an icon property."
+      );
+    }
+  }
+
+  render() {
+    const {
+      className,
+      small,
+      large,
+      style,
+      onClick,
+      icon
+    } = this.props;
+
+    let iconClasses = cx({
+      "large": large,
+      "small": small,
+      "icon": true
+    });
+
+    const iconContainer = icon ? <i style={style} className={icon}/> : null;
+
+    return (
+      <span className={cx(className)} onClick={onClick} styleName={iconClasses} style={style} >
```

Comment:
could className, onClick, and styleName be moved to the element in iconContainer to get rid of this wrapper span?